src/glossary.ts

```diff
@@ -73,7 +74,11 @@ export type ManyOf<ModelName extends KeyType> = RelationDefinition<
 
 export type ModelDefinition = Record<
   string,
-  (() => BaseTypes) | OneOf<any> | ManyOf<any> | PrimaryKeyDeclaration
+  | (() => BaseTypes)
+  | (() => BaseTypesArray)
```

Comment:
Do you see any practical benefit of using a separate type than specifying the `BaseTypes[]` directly here?

```ts
(() => BaseTypes | BaseTypes[])
```